jax_galsim/interpolatedimage.py

```diff
@@ -0,0 +1,1042 @@
+import copy
+import math
+import textwrap
+from functools import partial
+
+import galsim as _galsim
+import jax
+import jax.numpy as jnp
+from galsim.errors import (
+    GalSimIncompatibleValuesError,
+    GalSimRangeError,
+    GalSimUndefinedBoundsError,
+    GalSimValueError,
+)
+from galsim.utilities import doc_inherit
+from jax._src.numpy.util import _wraps
+from jax.tree_util import register_pytree_node_class
+
+from jax_galsim import fits
+from jax_galsim.bounds import BoundsI
+from jax_galsim.core.draw import draw_by_kValue, draw_by_xValue
+from jax_galsim.core.utils import compute_major_minor_from_jacobian, ensure_hashable
+from jax_galsim.gsobject import GSObject
+from jax_galsim.gsparams import GSParams
+from jax_galsim.image import Image
+from jax_galsim.interpolant import Quintic
+from jax_galsim.position import PositionD
+from jax_galsim.transform import Transformation
+from jax_galsim.utilities import convert_interpolant
+from jax_galsim.wcs import BaseWCS, PixelScale
+
+_KEYS_TO_REMOVE = [
```

Comment:
```suggestion
# These keys are removed from the public API of
# InterpolatedImage so that it matches the galsim
# one.
# The DirMeta class does this along with the changes to
# __getattribute__ and __dir__ below.
_KEYS_TO_REMOVE = [
```